src/inmanta/data/__init__.py

```diff
@@ -4256,66 +4261,41 @@ async def get_resource_events(
 
         # These two variables are actually of type datetime.datetime
         # but mypy doesn't know as they come from the DB
         # mypy also doesn't care, because they go back into the DB
-        current_deploy_start: object
         last_deploy_start: Optional[object]
 
-        end_query = """
-        with
-            base_ra as (
-            SELECT ra.*
-                FROM public.resourceaction_resource as jt
-                    INNER JOIN public.resourceaction as ra
-                        ON ra.action_id = jt.resource_action_id
-                    WHERE jt.environment=$1 AND ra.environment=$1 AND jt.resource_id=$2::varchar AND ra.action='deploy'
-                    ORDER BY ra.started DESC
-            )
-        SELECT
-            (SELECT started from base_ra ORDER BY started DESC LIMIT 1) as begin_started,
-            (SELECT status from base_ra ORDER BY started DESC LIMIT 1) as begin_status,
-            COALESCE((SELECT started from base_ra where status='deployed' ORDER BY started DESC LIMIT 1), NULL) as started
-        """
-
         async with cls.get_connection() as connection:
-            result = await connection.fetchrow(end_query, env.id, resource_id_str)
-
-            if not result or result["begin_status"] is None:
-                raise BadRequest(
-                    "Fetching resource events only makes sense when the resource is currently deploying. Resource"
-                    f" {resource_version_id_str} has not started deploying yet."
-                )
-            if result["begin_status"] != const.ResourceState.deploying:
-                raise BadRequest(
-                    "Fetching resource events only makes sense when the resource is currently deploying. Current deploy state"
-                    f" for resource {resource_version_id_str} is {result['begin_status']}."
-                )
-            current_deploy_start = result["begin_started"]
-            last_deploy_start = result["started"]
-
-            # Step3: Get the resource
+            # Step 1: Get the resource
+            # also check we are currently deploying
             resource: Optional[Resource] = await Resource.get_one(
                 environment=env.id, resource_id=resource_id_str, model=resource_id.version, connection=connection
             )
             if resource is None:
                 raise NotFound(f"Resource with id {resource_version_id_str} was not found in environment {env.id}")
 
+            if resource.status != const.ResourceState.deploying:
+                raise BadRequest(
+                    "Fetching resource events only makes sense when the resource is currently deploying. Current deploy state"
+                    f" for resource {resource_version_id_str} is {resource.status}."
+                )
+
+            # steps 2:
+            # find the interval between the current deploy (now) and the previous successful deploy
+            last_deploy_start = resource.last_success
```

Comment:
1. Might `last_successful_execute` be a better name? To make clear that it's not just about success "due to known good state" but explicit success due to handler execution? Completely optional suggestion.
2. As discussed, if we make this the metric to determine the window within which we look at events, it should be a part of these methods' contract/docstring and the handler's reload documentation. Perhaps with an explicit clarification that this means that a handler that uses these methods to monitor or respond to dependencies' events should mark its resource as deployed iff it successfully and completely processed all current events.

Reply:
for point 2, I added some docs to the relevant api endpoint, but it was already quite good

for reload, I'll wait for https://github.com/inmanta/inmanta-core/pull/6246

Reply:
> I added some docs to the relevant api endpoint

Did you push this?

Reply:
I did not